esnet/simple-lookup-service
# Elasticsearch db backend to 3.0.0 (#77)

* complete instance, constructors, init, queryAndPublish, deleteByRecord, getRecordByURI, deleteAllRecords

* added updateService with tests

* complete publish service, bulkUpdate; need to write tests

* add documentation to bulkUpdate

* add publishService tests

* created test for bulk update

* Delete expired records completed with tests

* find records in Range done with test

* add documentation to insert meethod

* add documentation for add timestamp method

* complete registeration APi and add gitignore

* fix server running

* Working renew get and delete

* made print statements into logs

* cleanup code

* fix existing

* add Databse connection keys

* update register service to google java format

* change ServiceElasticSearchTest to google java format

* add Database connection keys to accesss service

* add DataaseKeys to access service

* fix curl URI

* Record resources to google java format

* implwmwnt getkey

* remove old register service

* add documentation to tests and add test for keys

* create curl function for get and renew

* curl delete test compete

* add CURL test for adding reccord to databse

* add tests for getKey

* switch to yaml config file

* fix creating  indexg index if not exists

* Fix mantainance Job for elastic

* make changes in invoker for Maintenance job

* optimize exists to check all same type

* read config only once

* remove logs attempt

* remove all log files

* make connection an object; Exist optimization stopped working

* suggested changes made

* add basic initialization

* add bulkrenew tests

* add comments to connectDb

* fix editService comments

* fix deleteNotExistURI test'
'

* optimize exists

* fix memory management in invoker

* thread active fixes

* fix renew thread leak bug

* Query function All working

* complete query for any

* add regex function

* up to date

* fixed thread leak issue

* add bulkregister service for faster inserting of records

* bulk renew services working partially

* fix json parsing

* fix error due to existing documents

* add TTl field to JSON

* add api for bulk

* make changes for grafana

* fix returniong html

* making changes to work with grafana or kibana

* undo changes creating new branch for testing

* add locust tests

* Updating spec file

* Updating pom version numbers

* Changing search api to scroll api to support large requests. Sharing elasticsearch client between requests. Updated locust load test and pom version number

* Fixing typo in spec file

* Updating spec file

* Fixing spec file issues

* Further fixing spec file issues

* Fixing further issues with spec file

* Missing apache file definition

* Remove unwanted print statement

* Fix query method in elasticsearch to handle keyword filters

* Update spec file

* Cleaning up client-examples

* Cleaning up elasticsearch tests

* Cleaning up api classes

Co-authored-by: Kartik Arora <kartik@cs-it-6948080.dhcp.lbnl.us>
Co-authored-by: sowmyab <sowmyabalasubramanian@lbl.gov>
Co-authored-by: Sowmya Balasubramanian <sowmyab@users.noreply.github.com>
